klog_test.go

```diff
@@ -1447,6 +1447,110 @@ func TestErrorSWithLogr(t *testing.T) {
 	}
 }
 
+func TestCallDepthLogr(t *testing.T) {
+	logger := &callDepthTestLogr{}
+	logger.resetCallDepth()
+
+	testCases := []struct {
+		name  string
+		logFn func()
+	}{
+		{
+			name:  "Info log",
+			logFn: func() { Info("info log") },
+		},
+		{
+			name:  "InfoDepth log",
+			logFn: func() { InfoDepth(0, "infodepth log") },
+		},
+		{
+			name:  "InfoSDepth log",
+			logFn: func() { InfoSDepth(0, "infoSDepth log") },
+		},
+		{
+			name:  "Warning log",
+			logFn: func() { Warning("warning log") },
+		},
+		{
+			name:  "WarningDepth log",
+			logFn: func() { WarningDepth(0, "warningdepth log") },
+		},
+		{
+			name:  "Error log",
+			logFn: func() { Error("error log") },
+		},
+		{
+			name:  "ErrorDepth log",
+			logFn: func() { ErrorDepth(0, "errordepth log") },
+		},
+		{
+			name:  "ErrorSDepth log",
+			logFn: func() { ErrorSDepth(0, errors.New("some error"), "errorSDepth log") },
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			SetLogger(logger)
+			defer SetLogger(nil)
+			defer logger.reset()
+			defer logger.resetCallDepth()
+
+			// Keep these lines together.
+			_, wantFile, wantLine, _ := runtime.Caller(0)
+			tc.logFn()
+			wantLine++
```

Comment:
Something (literally) doesn't add up here. The actual logging calls, like `Error` in line 1480, have the anonymous function as their callsite. Why is the test expecting that this wrapper function gets skipped?

Here's the backtrace for the "Error log" test case:
```
0  0x00000000005a3512 in k8s.io/klog/v2.(*callDepthTestLogr).Error
   at ./klog_test.go:1632
1  0x0000000000587702 in k8s.io/klog/v2.(*loggingT).output
   at ./klog.go:912
2  0x000000000058575d in k8s.io/klog/v2.(*loggingT).printDepth
   at ./klog.go:733
3  0x0000000000585435 in k8s.io/klog/v2.(*loggingT).print
   at ./klog.go:715
4  0x000000000058b947 in k8s.io/klog/v2.Error
   at ./klog.go:1443
5  0x00000000005a7dbd in k8s.io/klog/v2.TestCallDepthLogr.func6
   at ./klog_test.go:1480
6  0x00000000005a8124 in k8s.io/klog/v2.TestCallDepthLogr.func9
   at ./klog_test.go:1501
```

I would expect `TestCallDepthLogr.func6` in line 1480 to be logged.

But the test logger then skips 6 frames and logs line 1501.

Okay, I think I understand. The test logger *intentionally* logs the "wrong" call site to enable writing tests like this.

Can you add a comment in the test logger that explains that? It is not obvious.

Reply:
are these comments sufficient?

https://github.com/kubernetes/klog/blob/2f2fd5b17f08fc1f6df20b51ebe9f6f014c63d8f/klog_test.go#L1671-L1672

https://github.com/kubernetes/klog/blob/2f2fd5b17f08fc1f6df20b51ebe9f6f014c63d8f/klog_test.go#L1658-L1659

Reply:
Yes.